### ISSUE TITLE
Regarding covariance matrix updates

### ISSUE THREAD

**Issue by dipamgoswami** [drimpossible/RanDumb#1]
Hi,
Thanks for sharing the code of your work.

If I understand the code correctly, you are not updating the covariance matrix in every task and assume that data from all the classes is available at the same time for computing the covariance matrix. In a continual setting, you have access to only the data from the current task, and thus one needs to update the matrix for every task. 

Is it fair to compare RanDumb with methods that compute a covariance matrix/gram matrix in every task and then update the matrix continually in new tasks since RanDumb is not implemented in a continual manner? For instance, in Table 2 of your paper, the covariance matrix will be different if you compute it for the initial 2 classes first and then keep updating it for one new class at a time compared to computing it for all classes in the dataset. I am not sure how much that will impact the performance.

Can you comment on this? Let me know if I am wrong somewhere.

**Comment by drimpossible** [comment]
Hi,

Updating the covariance matrix is a rank-1 update which is inherently an online operation so there should not be differences between the two. I have referenced the online version of the SLDA code alongside, which does the updates online (but is slow). You could compare the two and verify yourself that the performance is extremely similar!

This offline update is done *purely* because I had a lot of experiments to run when developing this idea, so wanted to not wait a few hours to get a single run done.

**Comment by drimpossible** [comment]
Heya Dipam,

We use empirical covariance matrix, which can be updated sample-wise with rank-1 update so there should not be a difference between online and offline updates to the covariance matrix. Clarified it in the code, and replaced the OAS covariance estimator to empricial covariance estimator (which was an accident for release version used to make the code similar to Hayes et.al.) 

The results should be very similar, let me know if they are different for the online and offline version of the code. The online version is really slow and takes a long time to run. I had originally used the online-version, but wanted to scale up experiments so started using the offline estimate as it was simply far faster, but we verified that we got the same results! 

Feel free to double check this yourself with the link given in `methods.py`, as they say trust but verify if you get anything different! Overall, RanDumb as given in the paper, should be a simple one-sample-at-a-time algorithm.

**Comment by drimpossible** [comment]
Heya Dipam,

We use empirical covariance matrix, which can be updated sample-wise with rank-1 update so there should not be a difference between online and offline updates to the covariance matrix. Clarified it in the code!

The results should be very similar, let me know if they are different for the online and offline version of the code. The online version is really slow and takes a long time to run. I had originally used the online-version, but wanted to scale up experiments so started using the offline estimate as it was simply far faster, but we verified that we got the same results!

Feel free to double check this yourself with the link given in methods.py! Overall, RanDumb as given in the paper, should be a simple one-sample-at-a-time algorithm.
